sway/tree/view.c

```diff
@@ -625,6 +625,9 @@ static void handle_foreign_activate_request(
 	struct sway_seat *seat;
 	wl_list_for_each(seat, &server.input->seats, link) {
 		if (seat->wlr_seat == event->seat) {
+			if (container_is_scratchpad_hidden_or_child(view->container)) {
```

Comment:
Since this is being edit anyway, do you mind adding the following:
```c
seat_consider_warp_to_focus(seat);
container_raise_floating(view->container);
```

So that this block matches the one in `cmd_focus`:
https://github.com/swaywm/sway/blob/4f718e6c75d2168d417b4b41c0e3d5408e0afded/sway/commands/focus.c#L368-L373

Reply:
Sounds good. Done.